server/controllers/request.js

```diff
@@ -0,0 +1,51 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler")
+
+// @route POST /requests/create
+// @desc Create new request
+// @access Public
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const request = await Request.create(req.body);
+
+  if (request) {
+    res.status(200).json({
+      success: {
+        request: {
+          ...request
+        }
+      }
+    })
+  }
+});
+
+// @route PUT /requests/edit/:requestId
+// @desc edit request
+// @access Public
+exports.editRequest = asyncHandler(async (req, res, next) => {
+  const request = await Request.findById(req.params.requestId)
+  
+  if (!request) {
+    res.status(404);
+    throw new Error("Request doesn't exist");
+  }
+  request.set(req.body);
```

Comment:
We would want only the sitter to edit a request or restrict the owner from editing fields like paid, accepted et cetera :)